### PR TITLE
Type uni

### PR DESCRIPTION
**Pull request by samgermain**
Solves most of https://github.com/ccxt/ccxt/issues/11121

Binance is still `future` and `delivery`
Bybit is still `future` `linear` and `inverse`

I'm willing to switch them, but just let me know when and how to do it for backwards compatibility

### CODE REVIEW

**Comment by frosty00** [comment]
looks good, i'll do okex and binance unification since they are more difficult

File: js/kucoinfutures.js [hunk 1]
```diff
@@ -1304,7 +1304,7 @@ module.exports = class kucoinfutures extends kucoin {
             'datetime': this.iso8601 (timestamp),
             'currency': code,
             'amount': amount,
-            'fromAccount': 'futures',
+            'fromAccount': 'future',
```
Comment (frosty00) [inline js/kucoinfutures.js:1307]: i don't think we should change this instance, let's only change the `.market` attributes

**Comment by frosty00** [comment]
can you revert all of `js/okex.js`?

**Comment by frosty00** [comment]
you haven't tested this PR and ik that because `okex` won't work with those edits

**Comment by frosty00** [comment]
let's only change the attributes inside of `.market`, don't touch any defaultTypes or other things then we can merge

**Comment by samgermain** [comment]
> 

no i usually just let `continuous-integration/appveyor/pr` and `Travis CI - Pull Request ` test it because it takes about 10 minutes locally and I have to wait through that before switching branches or anything

**Comment by samgermain** [comment]
> let's only change the attributes inside of `.market`, don't touch any defaultTypes or other things then we can merge

Why don't we want to unify any of that?

File: js/kucoinfutures.js (hunk 1 above)
Comment (samgermain) [reply js/kucoinfutures.js:1307]: I think we should use one term throughout

**Comment by frosty00** [comment]
> > let's only change the attributes inside of `.market`, don't touch any defaultTypes or other things then we can merge
> 
> Why don't we want to unify any of that?

because you can't do it in one go without breaking other things, for example how you have attempted to do in `js/okex`

File: js/kucoinfutures.js (hunk 1 above)
Comment (samgermain) [reply js/kucoinfutures.js:1307]: but I can add `fromAccount !== 'futures'` to the if

**Comment by frosty00** [comment]
> no i usually just let continuous-integration/appveyor/pr and Travis CI - Pull Request test it because it takes about 10 minutes locally and I have to wait through that before switching branches or anything

That's not actually testing the code entirely tho, I suggest to actually change the defaultType and call each method everytime you change something critical like that to see if it actually works

File: js/kucoinfutures.js (hunk 1 above)
Comment (frosty00) [reply js/kucoinfutures.js:1307]: > I think we should use one term throughout

I agree, but we can't change it all at once without introducing bugs, so first let's try to unify the markets and then we unify the defaultTypes and the accountTypes

**Comment by frosty00** [comment]
i'd also suggest to make one PR per exchange so it is easier to merge

**Pull request closed without merging**